binary_and_decimal.rb

```diff
@@ -1,27 +1,37 @@
-# 1. Write a method named `binary_to_decimal` that creates an array of size 8.
-#    Fill the array randomly with 0’s and 1’s. Print out the array so that it
-#    appears as a binary number. Calculate the decimal value for this binary
-#    number using the algorithm you devised in class. Print out the decimal value.
 def binary_to_decimal
-  puts "Not implemented"
+  bin = Array.new(8){rand(0..1)}
+
+  reverse_bin = bin.reverse
+  i = 0
+  add_nums = []
+  reverse_bin.each do |digit|
+    add_nums << digit * (2 ** i)
+    i += 1
+  end
+  
+  decimal_num = add_nums.reduce(0){|sum, num| sum + num}
+
+  return "The binary ##{bin.join} equals the decimal ##{decimal_num}"
 end
 
-# 2. Write a method named `array_equals` that accepts two integer arrays as
-#    parameters. The method return `true` if the arrays contain the same elements
-#    in the same order, and returns `false` otherwise.
-# Note: Do not use Array class methods for comparing the whole array at once.
-#       You may use array indexing to retrieve one element at a time,
-#       compare individual elements with each other and
-#       you may retrieve the length of an array.
 def array_equals(array1, array2)
-  puts "Not implemented"
-  return true
-end
+  if array1.length != array2.length
+    return false
+  end
 
+  i = 0
+  array1.each do |char|
+    if char != array2[i]
+      return false
+    else
+      return true # (if array contains the same elements in same order)
```

Comment:
You are returning after checking only the first element. If the first element is the same, you want to continue checking the next by incrementing `i` and continuing in the loop. Then, outside the loop, return true.